zebra-rpc/tests/serialization_tests.rs

```diff
@@ -0,0 +1,763 @@
+  "bits": "1c023081",
+  "difficulty": 61301397.633212306,
+  "previousblockhash": "0000000000d12367f80be78e624d263faa6e6fda718453cbb6f7dc71205af574",
+  "nextblockhash": "0000000001d8a2a9c19bc98ecb856c8406ba0b2d7d42654369014e2a14dd9c1d"
+}
+"#;
+    let r: GetBlockHeader = serde_json::from_str(json)?;
+
+    let GetBlockHeader::Object(obj) = &r else {
+        panic!("Expected Object variant");
+    };
+
+    let new_obj = GetBlockHeader::Object(Box::new(GetBlockHeaderObject {
+        hash: obj.hash,
+        confirmations: obj.confirmations,
+        height: obj.height,
+        version: obj.version,
+        merkle_root: obj.merkle_root,
+        block_commitments: obj.block_commitments,
+        final_sapling_root: obj.final_sapling_root,
+        sapling_tree_size: obj.sapling_tree_size,
+        time: obj.time,
+        nonce: obj.nonce,
+        solution: obj.solution,
+        bits: obj.bits,
+        difficulty: obj.difficulty,
+        previous_block_hash: obj.previous_block_hash,
+        next_block_hash: obj.next_block_hash,
+    }));
+
+    assert_eq!(r, new_obj);
+
+    Ok(())
+}
+
+#[test]
+fn test_get_block_height_hash() -> Result<(), Box<dyn std::error::Error>> {
+    let json = r#"
+{"height":2931705,"hash":[35,5,244,118,21,236,8,168,3,119,95,171,238,9,233,152,250,106,153,253,6,176,155,7,155,161,146,1,0,0,0,0]}
+"#;
+    let obj: GetBlockHeightAndHash = serde_json::from_str(json)?;
+    let new_obj = GetBlockHeightAndHash {
+        height: obj.height,
+        hash: obj.hash,
+    };
+
+    assert_eq!(obj, new_obj);
```

Comment:
This assertion isn't really testing anything, could we do a round-trip serialization test instead?

Reply:
The goal is to ensure clients can parse the JSON and read all the contents of the response. If we are able to reconstruct the object, then it means we have succeed that goal.

To complete the round-trip we'd need to serialize and canonicalize it, and then canonicalize the test vectors... But I think we already test serialization with the insta test vectors